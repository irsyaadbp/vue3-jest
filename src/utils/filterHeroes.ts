// filter.js
const heroes = [
  //   { name: "Dynaguy", powers: ["fly", "disintegration ray"] },
  //   { name: "Apogee", powers: ["fly", "gravity control"] },
  {
    name: "Blazestone",
    powers: ["control of fire", "pyrotechnic dischargers"],
  },
  { name: "Frozone", powers: ["freeze water"] },
  { name: "Mr. Incridible", powers: ["phyisical strength"] },
  { name: "Elastigirl", powers: ["phyisical strength"] },
  { name: "Violet ", powers: ["invisibility", "force fields"] },
  { name: "Dash", powers: ["speed"] },
  { name: "Jack", powers: ["shapeshifting", "fly"] },
];

function showBudi(condition: boolean) {
  if (condition) return console.log("budi");
}

function getFlyingSuperHeroes() {
  return heroes.filter((hero) => {
    return hero.powers.includes("fly");
  });
}

export { getFlyingSuperHeroes, showBudi };
