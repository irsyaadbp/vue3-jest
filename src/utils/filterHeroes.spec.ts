import { getFlyingSuperHeroes } from "./filterHeroes";

describe("getFlyingSuperHeroes", () => {
  it("should be get flying super heroes", () => {
    const flyingSuperHeroes = getFlyingSuperHeroes();
    // console.log(flyingSuperHeroes);
    // expect(flyingSuperHeroes).toEqual([
    //   { name: "Dynaguy", powers: ["fly", "disintegration ray"] },
    //   { name: "Apogee", powers: ["fly", "gravity control"] },
    //   { name: "Jack", powers: ["shapeshifting", "fly"] },
    // ]);
    expect(flyingSuperHeroes).toMatchSnapshot();
  });

  it("should be get flying super heroes", () => {
    const flyingSuperHeroes = getFlyingSuperHeroes();
    // console.log(flyingSuperHeroes);
    // expect(flyingSuperHeroes).toEqual([
    //   { name: "Dynaguy", powers: ["fly", "disintegration ray"] },
    //   { name: "Apogee", powers: ["fly", "gravity control"] },
    //   { name: "Jack", powers: ["shapeshifting", "fly"] },
    // ]);
    expect(flyingSuperHeroes).toMatchInlineSnapshot(`
[
  {
    "name": "Jack",
    "powers": [
      "shapeshifting",
      "fly",
    ],
  },
]
`);
  });
});
