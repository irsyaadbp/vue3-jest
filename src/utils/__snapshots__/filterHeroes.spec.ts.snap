// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFlyingSuperHeroes should be get flying super heroes 1`] = `
[
  {
    "name": "Jack",
    "powers": [
      "shapeshifting",
      "fly",
    ],
  },
]
`;
